repose-aggregator/components/filters/valkyrie-authorization/src/main/scala/org/openrepose/filters/valkyrieauthorization/ValkyrieAuthorizationFilter.scala

```diff
@@ -136,10 +136,11 @@ class ValkyrieAuthorizationFilter @Inject()(configurationService: ConfigurationS
     def authorizeDevice(permissions: ValkyrieResult, deviceId: Option[String]): ValkyrieResult = {
       def authorize(deviceId: String, permissions: UserPermissions, method: String): ValkyrieResult = {
```

Comment:
Since the authorize function is nested in the authorizeDevice function, it's pretty sketchy to use the same parameter name ("permissions") for both.